internal/importers/glock/importer.go

```diff
@@ -110,18 +116,22 @@ func parseGlockLine(line string) (*glockPackage, error) {
 	}, nil
 }
 
-func (g *Importer) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+func (g *Importer) convert(pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock) {
 	g.Logger.Println("Converting from GLOCKFILE ...")
 
 	packages := make([]base.ImportedPackage, 0, len(g.packages))
 	for _, pkg := range g.packages {
 		// Validate
 		if pkg.importPath == "" {
-			return nil, nil, errors.New("invalid glock configuration, import path is required")
+			g.Logger.Println("  Warning: Invalid glock configuration, import path is required")
+			continue
 		}
 
 		if pkg.revision == "" {
-			return nil, nil, errors.New("invalid glock configuration, revision is required")
+			g.Logger.Printf(
```

Comment:
I don't quite understand. If the revision isn't present we would just not have the `LockHint`. Isn't it better to still add the package? I would also need to do these changes in a few other places where I continue with importing the package even though the contraint/lock hint could not be resolved.

Reply:
glock is unique because it doesn't provide a ConstraintHint or Source, just the LockHint. So if the imported config have a LockHint, then there's no point it continuing. Dep doesn't add "empty constraints" to the manifest during import, and the normal solve step will add the project to the lock if needed, so when the LockHint, ConstraintHint and Source are empty, it's a no-op.

The other tools may still have a ConstraintHint or Source, which is why we don't stop as soon as we know that the LockHint is empty.

Reply:
Okay. I looked through the rest of the importers and the same scenario applies to `govend` as well:
https://github.com/golang/dep/blob/c77daf665f7803c6ca1e2835637f2ed8eb5e9780/internal/importers/govend/importer.go#L100-L109

Additionally, I do have a concern regarding this because the warning might be misleading. Like me, the user might also be confused by `Skipping package..`. How about changing the warning to something like `Warning: Skipping import with empty constraints. The solve step will add the dependency to the lock if needed.`? too long?